File: batch850/SerivceExample/app/src/main/java/com/example/serivceexample/MyBoundedService.java
package com.example.serivceexample;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Binder;
import android.os.IBinder;

import androidx.annotation.Nullable;

public class MyBoundedService extends Service {

    MediaPlayer mediaPlayer;

    private final IBinder myBinder = new MyLocalBinde();

    boolean res =false;
    String start;

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return myBinder;
    }

    public MyBoundedService() {
    }

    public class MyLocalBinde extends Binder{
     MyBoundedService  getService()
        {
            return MyBoundedService.this;
        }
    }

    public void startmedia()
    {
        mediaPlayer = MediaPlayer.create(this,R.raw.song);
        mediaPlayer.start();

    }
    public void pausemedia()
    {
        if( res == true)
        {
            mediaPlayer.pause();
            res = false;
        }else
        {
            mediaPlayer.start();
            res = true;
        }
    }
    public void reset(){
        mediaPlayer.stop();
        mediaPlayer.release();
    }
}

File: Rahoul3pm/BoundedServiceKotlin/app/src/main/java/com/example/boundedservicekotlin/MyBoundedService.kt
package com.example.boundedservicekotlin

import android.app.Service
import android.content.Intent
import android.media.MediaPlayer
import android.os.Binder
import android.os.IBinder

class MyBoundedService(): Service() {
    lateinit var mediaPlayer: MediaPlayer
    var res = false



    private val myBinder: IBinder = MyBounderClass()

    override fun onBind(p0: Intent?): IBinder? {
          return myBinder

    }

    fun startsone()
    {
mediaPlayer = MediaPlayer.create(applicationContext,R.raw.song)
        mediaPlayer.start()
    }

    fun paussong()
    {
        res = if (res == true) {
            mediaPlayer.pause()
            false
        } else {
            mediaPlayer.start()
            true
        }
    }

    fun resetsong()
    {

        mediaPlayer.stop()
        mediaPlayer.release()
    }

inner class MyBounderClass :Binder(){
    fun getbinder():MyBoundedService{
              return  this@MyBoundedService
    }
}
}

